packages/editor-ui/src/stores/nodeCreator.ts

```diff
@@ -308,35 +308,45 @@ export const useNodeCreatorStore = defineStore(STORES.NODE_CREATOR, {
 			return nodesWithActions;
 		},
 		mergedAppNodes(): INodeTypeDescription[] {
-			const mergedNodes = [...this.visibleNodesWithActions]
-				// Sort triggers so they are always on top and when later get merged
-				// they won't be discarded if they have the same name as a core node which doesn't contain actions
-				.sort((a, b) => {
-					if (a.group.includes('trigger')) return -1;
-					if (b.group.includes('trigger')) return 1;
-
-					return 0;
-				})
-				.reduce((acc: Record<string, INodeTypeDescription>, node: INodeTypeDescription) => {
-					const clonedNode = deepCopy(node);
-					const actions = node.actions || [];
-
-					const normalizedName = node.name.toLowerCase().replace('trigger', '');
-					const existingNode = acc[normalizedName];
-
-					if (existingNode) existingNode.actions?.push(...actions);
-					else acc[normalizedName] = clonedNode;
-
-					acc[normalizedName].displayName = node.displayName.replace('Trigger', '');
-
-					return acc;
-				}, {});
+			const triggers = this.visibleNodesWithActions.filter((node) =>
+				node.group.includes('trigger'),
+			);
+			const apps = this.visibleNodesWithActions
+				.filter((node) => !node.group.includes('trigger'))
+				.map((node) => {
+					const newNode = deepCopy(node);
+					newNode.actions = newNode.actions || [];
+					return newNode;
+				});
+
+			triggers.forEach((node) => {
+				const normalizedName = node.name.toLowerCase().replace('trigger', '');
+				const app = apps.find((node) => node.name.toLowerCase() === normalizedName);
+				const newNode = deepCopy(node);
+				if (app && app.actions?.length) {
+					app?.actions?.push(...(newNode.actions || []));
+					app.description = newNode.description; // default to trigger description
+				} else {
+					newNode.actions = newNode.actions || [];
+					apps.push(newNode);
+				}
+			});
 
-			const filteredNodes = Object.values(mergedNodes).map((node) => ({
+			const filteredNodes = apps.map((node) => ({
 				...node,
 				actions: filterActions(node.actions || []),
 			}));
 
+			filteredNodes.sort((a, b) => {
```

Comment:
I think this might not be necessary anymore, since we do displayName sorting as part of `transformCreateElements` in `NodeCreator/MainPanel.vue`

Reply:
ya was not sure.. will remove and test